### PR TITLE
Get default aws subnet ids 

### LINKED ISSUE #1130: Getting default subnets of default VPC
The VPC object obtained when running `aws.GetDefaultVpc` includes all subnets in the default VPC, and not just the default subnets. This is a problem because often this is used with the intention to get the default subnets in a region, as by default the default VPC has a default subnet in each availability zone. However, there are occasions where more subnets are created in the default VPC causing unintended effects. 

Related: 
* [cloud nuke tests create subnet in default VPC](https://github.com/gruntwork-io/cloud-nuke/blob/a696846eadb65719badc5bc56706a9c8a9b91292/aws/eks_utils_for_test.go#L188)
* [terratest test `TestGetVpcsE` sometimes fails due to unmatching AZ & subnet number](https://app.circleci.com/pipelines/github/gruntwork-io/terratest/993/workflows/ad532d40-0119-4302-978e-7693ba31b5f5/jobs/7638/tests)
* [terraform-aws-asg tests sometimes fail because the ELB cannot be attached to a list of subnets that has more than one subnet in the same AZ](https://github.com/gruntwork-io/terraform-aws-asg/issues/151)

I'll be opening a PR to address this soon.

### PR DESCRIPTION
**Pull request by Etiene**
## Description

Resolves #1130.

The VPC object obtained when running aws.GetDefaultVpc includes all subnets in the default VPC, and not just the default subnets. This is a problem because often this is used with the intention to get the default subnets in a region, as by default the default VPC has a default subnet in each availability zone. However, there are occasions where more subnets are created in the default VPC causing unintended effects.

For example: 
* [cloud nuke tests create subnet in default VPC](https://github.com/gruntwork-io/cloud-nuke/blob/a696846eadb65719badc5bc56706a9c8a9b91292/aws/eks_utils_for_test.go#L188)
* [terratest test `TestGetVpcsE` sometimes fails due to unmatching AZ & subnet number](https://app.circleci.com/pipelines/github/gruntwork-io/terratest/993/workflows/ad532d40-0119-4302-978e-7693ba31b5f5/jobs/7638/tests)
* [terraform-aws-asg tests sometimes fail because the ELB cannot be attached to a list of subnets that has more than one subnet in the same AZ](https://github.com/gruntwork-io/terraform-aws-asg/issues/151)

This PR 
* adds `aws.GetDefaultSubnetIDsForVpc` to retrieve default subnet ids
* updates `aws.IsPublicSubnet` to check implicit association with the main route table in VPC when the subnet does not have any explicitly associated route tables
* and fixes the `TestGetVpcsE` test to check for greater or equal number of subnets and availability zones instead of equal. 

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes 

* Added `aws.GetDefaultSubnetIDsForVpc` to retrieve default subnet ids
* Updated `aws.IsPublicSubnet` to check implicit association with the main route table in VPC when the subnet does not have any explicitly associated route tables

### CODE REVIEW

File: modules/aws/vpc.go [hunk 1]
```diff
@@ -182,6 +184,34 @@ func GetTagsForVpcE(t testing.TestingT, vpcID string, region string) (map[string
...
+		return nil, fmt.Errorf("Only default VPCs have default subnets but VPC with id %s is not default VPC", vpc.Id)
+	}
+	subnetIDs := []string{}
+	numSubnets := len(vpc.Subnets)
+	if numSubnets == 0 {
+		return nil, fmt.Errorf("Expected to find at least one subnet in vpc with ID %s but found zero", vpc.Id)
+	}
+
+	for _, subnet := range vpc.Subnets {
+		if subnet.DefaultForAz == true {
```
Comment (denis256) [inline modules/aws/vpc.go:208]: I was wondering if is required to check for true

**Review by denis256 — APPROVED** [review]
:+1:

File: modules/aws/vpc.go (hunk 1 above)
Comment (Etiene) [reply modules/aws/vpc.go]: oh yea, this is not necessary. removed in [864adce](https://github.com/gruntwork-io/terratest/pull/1131/commits/864adceb724fb232d22618d45e4885db7c644234) and added a new check in the tests for recently created subnets too

**Review by denis256 — APPROVED** [review]

**Comment by Etiene** [comment]
Thanks for approving this @denis256!

PS: I don't have the powers to re-run the CI tests nor merge PRs in this repo :)

**Comment by yorinasub17** [comment]
Kicked off test after rebasing on `master` (which has a few test fixes): https://app.circleci.com/pipelines/github/gruntwork-io/terratest/1037/workflows/495242aa-7f88-4005-8776-1b5d26fefdfe

**Comment by yorinasub17** [comment]
Confirmed the test failures are unrelated to this change, so will go ahead and merge + release this!

**Pull request merged**